packages/core/src/search/search.ts

```diff
@@ -1,12 +1,14 @@
 import { Resource, ResourceType, SearchParameter } from '@medplum/fhirtypes';
 import { badRequest, OperationOutcomeError } from '../outcomes';
-import { TypedValue, stringifyTypedValue, globalSchema } from '../types';
+import { TypedValue, stringifyTypedValue, globalSchema, getSearchParameter } from '../types';
 import { evalFhirPathTyped } from '../fhirpath/parse';
+import { SearchParameterDetails, getSearchParameterDetails } from './details';
 
 export const DEFAULT_SEARCH_COUNT = 20;
 
 export interface SearchRequest<T extends Resource = Resource> {
   readonly resourceType: T['resourceType'];
+  chains?: ChainedSearchParameter[];
```

Comment:
I tugged on this thread a bit on my branch: in theory it would be possible to defer parsing the search chain and just leave it as `Filter` with the chained params in the `code` property.  This would require exposing more of the internal search parsing functions in `@medplum/core` as public, since e.g. `parseParameter()` is used to process the final link in the chain (which may contain modifiers etc that couldn't be parsed in the initial pass)